src/controllers/project.ts

```diff
@@ -0,0 +1,202 @@
+    lastCommitMessage,
+    // add lastCommitHash,
+    // add publicIpAddress,
+    webhookId,
+  });
+
+  await DeploymentModel.findByIdAndUpdate(newProject._id, {
+    buildingLog,
+    repoUpdatedAt,
+    deployedStatus: "test...",
+    lastCommitMessage: "test...",
+    lastCommitHash: "test...",
+  });
+
+  return;
+});
+
+export const getUserProjects = catchAsync(async (req, res, next) => {
+  const { user_id } = req.params;
+
+  if (!user_id) {
+    return next(createError(401, "Cannot find environment data 'user_id'"));
+  }
+
+  const userProjects = await UserModel.findByIdAndGetProjects(user_id);
+
+  const filteredUserProjects = userProjects?.map(project => {
+    const {
+      _id,
+      space,
+      repoName,
+      repoCloneUrl,
+      repoUpdatedAt,
+      projectName,
+      nodeVersion,
+      installCommand,
+      buildCommand,
+      buildType,
+      envList,
+      instanceId,
+      deployedUrl,
+      lastCommitMessage,
+      lastCommitHash,
+      deployments,
+    } = project;
+
+    const filteredProjectData = {
+      repoName,
+      space,
+      repoCloneUrl,
+      repoUpdatedAt,
+      projectName,
+      nodeVersion,
+      installCommand,
+      buildCommand,
+      envList,
+      buildType,
+      deployedUrl,
+      instanceId,
+      projectId: _id,
+      lastCommitMessage,
+      lastCommitHash,
+      deployments,
+    };
+
+    return filteredProjectData;
+  });
+
+  return res.json({
+    result: "ok",
+    data: filteredUserProjects,
+  });
+});
+
+export const updateProject = catchAsync(async (req, res, next) => {
+  const { action, pull_request, repository } = req.body;
+
+  if (action !== "closed" || !pull_request.merged) {
+    return next(
+      createError(
+        401,
+        "Received abnormal PR data or PR that has not yet been merged.",
+      ),
+    );
+  }
+
+  const githubAccessToken = Config.USER_CREDENTIAL_TOKEN;
+  const githubClient = new GithubClient(githubAccessToken as string);
+  const project = new ProjectService();
+
+  const repoOwner = repository.owner.login;
+  const repoName = repository.name;
+  const headRef = pull_request.head.ref;
+
+  const { commit } = await githubClient.getHeadCommitMessage(
+    repoOwner,
+    repoName,
+    headRef,
+  );
+
+  const redeployOptions = {
+    repoCloneUrl: repository.clone_url,
+    lastCommitMessage: commit.message,
+    repoName,
+  };
+  await project.redeployProject(redeployOptions);
+
+  return res.json({
+    result: "ok",
+  });
+});
+
+export const deleteProject = catchAsync(async (req, res, next) => {
+  const { githubAccessToken } = req.query;
+  const { projectId } = req.params;
+
+  if (!githubAccessToken || !projectId) {
+    return next(
+      createError(
+        400,
+        "Cannot find environment data 'githubAccessToken', and 'projectName'",
+      ),
+    );
+  }
+
+  const deletedProject = await ProjectModel.findByIdAndDelete(projectId);
+
+  if (!deletedProject) {
+    return next(createError(500));
```

Comment:
```suggestion
    return next(createError(500, "Error message..."));
```

나중에 에러 메시지 추가해주시면 감사하겠습니다..!

Reply:
고민하다가 500만 넣었는데 예리하시군요.. 
추가해놓겠습니다!